base/test/valveModule_tests.cpp

```diff
@@ -0,0 +1,330 @@
+    }
+    bool validateInputPins()
+    {
+        return true;
+    }
+};
+
+BOOST_AUTO_TEST_CASE(basic)
+{
+    auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+    auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::GENERAL));
+    auto pinId = source->addOutputPin(metadata);
+    auto valve = boost::shared_ptr<ValveModule>(new ValveModule(ValveModuleProps(2)));
+    source->setNext(valve, true, false);
+    auto sink = boost::shared_ptr<SinkModule>(new SinkModule(SinkModuleProps()));
+    valve->addOutputPin(metadata);
+    valve->setNext(sink, true, false);
+     
+
+    BOOST_TEST(source->init());
+    BOOST_TEST(valve->init());
+    BOOST_TEST(sink->init());
+
+    auto sinkQue = sink->getQue();
+
+    auto frame = source->makeFrame(1023, pinId);
+    frame_container frames;
+    frames.insert(make_pair(pinId, frame));
+   
+    // We are sending 4 frames with enable false so only no frames pass through the valve
+    for (int i = 0; i < 4; i++)
+    {
+        source->send(frames);
+        valve->step();
+    }
+
+    BOOST_TEST(sinkQue->size() == 0);
+    //We are resetting the props to 4 by command and enable true 
+    valve->allowFrames(4);
+    valve->step();
+
+    for (int i = 0; i < 4; i++)
+    {
+        source->send(frames);
+        valve->step();
+    }
+
+    BOOST_TEST(sinkQue->size() == 4);
+
+    //The props are changed by passing 2 in allowFrames()
+    valve->allowFrames(2);
+    valve->step();
+
+ 
+for (int i = 0; i < 2; i++)
+{
+    source->send(frames);
+    valve->step();
+}
+
+BOOST_TEST(sinkQue->size() == 6);
+
+//We are sending 4 frames again with enable false
+for (int i = 0; i < 4; i++)
+{
+    source->send(frames);
+    valve->step();
+}
+
+BOOST_TEST(sinkQue->size() == 6);
+
+// The props are changed by passing 0 in allowFrames()
+valve->allowFrames(0);
+valve->step();
+
+
+for (int i = 0; i < 2; i++)
+{
+    source->send(frames);
+    valve->step();
+}
+}
+
+BOOST_AUTO_TEST_CASE(multiple_pins)
+{
+    /*src - 2 output pins
+    valve module - sieve = disabled
+    2 children modules of valve */
+    auto source = boost::shared_ptr<ExternalSourceModule>(new ExternalSourceModule());
+    auto metadata = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::GENERAL));
+    auto metadata1 = framemetadata_sp(new FrameMetadata(FrameMetadata::FrameType::RAW_IMAGE));
+    auto pinId = source->addOutputPin(metadata);
+    auto pinId1 = source->addOutputPin(metadata1);
+
+    auto valve = boost::shared_ptr<ValveModule>(new ValveModule(ValveModuleProps(2)));
+    // always use disable sieve with valve module 
+    source->setNext(valve, true, false);
+
+    auto sink1 = boost::shared_ptr<SinkModule>(new SinkModule(SinkModuleProps()));
+    auto sink2 = boost::shared_ptr<SinkModule>(new SinkModule(SinkModuleProps()));
+    valve->addOutputPin(metadata);
+    valve->addOutputPin(metadata1);
+    valve->setNext(sink1, true, false);
```

Comment:
all outgoing connections of valve will need disabled sieve - so this is correct

Reply:
Also, this setNext should be overriden in the valveModule to allow valve->setNext(sinkModule)